### PR TITLE
#167484584 Generate and verify jwt token

### PR DESCRIPTION
**Pull request by Jkadhuwa**
**What does this PR do?**
Generates and verifies jwt token which will be used for authentication. 

**Description of Task to be completed?**
Add a helper function to generate jwt token
Add middleware to verify the supplied token

**How should this be manually tested?**
When a user successfully signs up or logs in should be given an authentication token

**What are the relevant pivotal tracker stories?**
[Generate and verify token ](https://www.pivotaltracker.com/story/show/167484584)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,7 +8,6 @@
     "coverage": "nyc npm test && nyc report --reporter=text-lcov | coveralls",
     "build": "rm -rf build && babel -d ./build ./src -s",
     "start": "npm run build && node ./build/index.js",
-    "dev": "nodemon --exec babel-node ./src/index.js",
```
Comment (mkiterian) [inline package.json:11]: why was this removed

File: src/helpers/auth.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import dotenv from 'dotenv';
+import jwt from 'jsonwebtoken';
+
+dotenv.config();
+
+/**
+ * @method GenToken
+ * - Generates jwtToken for authentication using user-id, user-email and role
+ * -
```
Comment (mkiterian) [inline src/helpers/auth.js]: what's this line for?

File: src/helpers/auth.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import dotenv from 'dotenv';
+import jwt from 'jsonwebtoken';
+
+dotenv.config();
+
+/**
+ * @method GenToken
```
Comment (mkiterian) [inline src/helpers/auth.js]: this is a function and not a class method. Also I the name here should be the exact match of the function name

File: src/middleware/verifyToken.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import jwt from 'jsonwebtoken';
+
+/**
+ * @method verifyToken
+ * - Gets token from req.params and verifies from requests
+ */
+
+const verifyToken = (req, res, next) => {
+  const token = req.headers.authorization;
+  if (!token || typeof token === 'undefined') {
```
Comment (mkiterian) [inline src/middleware/verifyToken.js]: do we need this part `typeof token === 'undefined'`? coz I'd expect  `!token` to evaluate to true for undefined

**Review by mkiterian — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (Jkadhuwa) [reply package.json:11]: We had two dev scripts.
